index.html

```diff
@@ -2,21 +2,130 @@
 <html lang="en">
   <head>
     <meta charset="UTF-8" />
-    <meta
-      name="viewport"
-      content="width=device-width, initial-scale=1, shrink-to-fit=no"
-    />
-    <title>My Blog</title>
-    <link
-      href="//fonts.googleapis.com/css?family=Roboto:400,500,300"
-      rel="stylesheet"
-      type="text/css"
-    />
+    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
     <link rel="stylesheet" href="css/style.css" />
-    <link rel="shortcut icon" type="image/x-icon" href="favicon.ico" />
+    <link rel="shortcut icon" type="image/x-icon" href="images/favicon.ico" />
+    <title>Nagas</title>
   </head>
   <body>
-    <!-- Add your HTML markup here -->
-    <!-- Remember: Use semantic HTML tags like <header>, <main>, <nav>, <footer>, <section> etc -->
+    <div id="app-container">
+      <!-- Semantically, I checked if <nav> MUST be within <header> but there is no hard rule -->
+      <!-- dev.w3.org/html5/spec/Overview.html#the-nav-element -->
+      <nav>
+        <h3 class="nav-title">Navigation</h3>
+        <ul class="nav-list">
+          <li class="nav-list-item"><a class="nav-link" href="#article-id-01">Article One</a></li>
+          <li class="nav-list-item"><a class="nav-link" href="#article-id-02">Article Two</a></li>
+          <li class="nav-list-item"><a class="nav-link" href="#article-id-03">Article Three</a></li>
+          <li class="nav-list-item"><a class="nav-link" href="#article-id-04">Article Four</a></li>
+        </ul>
+      </nav>
+      <header>
+        <div class="header-image-title-container">
+          <img class="header-image" src="images/logo01.png" alt="Naga Logo">
+          <h1 class="header-title">Nagas</h1>
+        </div>
+        <span class="header-description">
+          <strong>Naga</strong>, in Hinduism and Buddhism, is the Sanskrit word for a deity taking the form of a great snake, specifically the multi-headed king cobra. The naga which is seen sheltering the Buddha while meditating is known as Mucilanda.
+        </span>
+      </header>
+      <main>
+        <article id="article-id-01">
+          <h2 class="article-title">Mucalinda</h2>
+          <div class="article-figures-container">
+            <figure>
+              <img class="article-image" src="images/image01.jpg" alt="A painting of Mucalinda protecting Buddha">
+              <figcaption class="article-image-caption">Mucalinda protects Buddha (detail), Wat Kasattrathirat, Ayutthaya</figcaption>
+              <cite>
+                <a href="https://commons.wikimedia.org/wiki/File:090_Mucalinda_protects_Buddha_(detail)_(9192151900).jpg" target="_blank">citation</a>
+              </cite>
```

Comment:
The cite tag must contain the name of the word that it is referring to. Doing this will fix a second issue here of making the link discernible. Currently if you use a screen reader to list the links on the page it would list multiple with the value of "citation" the user would not be able to tell the difference